pkg/processor/k8sprocessor/pod_association.go

```diff
@@ -32,9 +32,16 @@ import (
 func extractPodID(ctx context.Context, attrs pdata.AttributeMap, associations []kube.Association) (podIdentifierKey string, podIdentifierValue kube.PodIdentifier) {
 	hostname := stringAttributeFromMap(attrs, conventions.AttributeHostName)
 	var connectionIP kube.PodIdentifier
-	if c, ok := client.FromContext(ctx); ok {
-		connectionIP = kube.PodIdentifier(c.IP)
+
+	if c := client.FromContext(ctx); c.Addr != nil {
```

Comment:
Interesting change, I believe it's not due to update of the core?

Reply:
I don't see it present in the upstream code. Should we put it there? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6fff7610fdf982db67904ee91006e63faf91a254/processor/k8sattributesprocessor/pod_association.go#L89-L96

Reply:
I believe this is the change

Reply:
It seems it will always be just the IP address though: https://github.com/open-telemetry/opentelemetry-collector/blob/ac9619e17e22da915b25dc75f0de61274c1760ce/client/client.go#L144-L162

Reply:
And there's some splitting already in-place: https://github.com/open-telemetry/opentelemetry-collector/blob/ac9619e17e22da915b25dc75f0de61274c1760ce/client/client.go#L144-L162

Reply:
@pmm-sumo Can you please check now.